index.md

```diff
@@ -144,14 +142,28 @@ If this isn't sufficient for your design, please seek out experts to talk to.
   AES-GCM is just AES-CTR (an unauthenticated mode) plus GMAC. This means that an attacker who has the key can just decrypt the ciphertext in AES-CTR mode and ignore the extra GMAC tag. That or they can use a library like OpenSSL which will release unverified plaintext (see prior point) and get the data that way.
 
 ### Signatures
-[Digital Signatures](https://en.wikipedia.org/wiki/Digital_signature) are generally safe to use, but many people assume they have properties that they do not. At the core all they mean is that *without the private key, an attacker cannot find a signature over an arbitrary value for which they don't already know a signature*. (a.k.a. [existential unforgeability])
+[Digital Signatures](https://en.wikipedia.org/wiki/Digital_signature) are generally safe to use, but many people assume they have properties that they do not. At their core all they mean is that *without the private key, an attacker cannot find a signature for the associated public key over an arbitrary value for which they don't already know a signature*. (a.k.a. [existential unforgeability])
+The best overview of these additional (unpromised) security properties is in  [Seems Legit: Automated Analysis of Subtle Attacks on Protocols that Use Signatures](https://eprint.iacr.org/2019/779)
+and I strongly recommend that everyone at least skim that paper.
+(There is lots of formal verification in there which you can probably skip, but the introduction, defined properties, and case studies are critical.)
+For an easier read, though focused entired on ECDSA, you should look at [How Not to Use ECDSA](https://yondon.blog/2019/01/01/how-not-to-use-ecdsa/).
+
+That said, here are lots of "gotchas" for digital signatures:
```

Comment:
Could add: "An attacker may be able to extract information from a signature, such as the the signer's public key or the hash of the signed message."

Reply:
I already have this:
> * Signatures do not (necessarily) hide the message they sign. They'll commonly reveal the hash (or more) of the data being signed.
> * Signatures do not hide the public key used to verify the message.
>   * "Public Key Recovery" is a standardized part of ECDSA. (See [SEC2](https://secg.org/sec1-v2.pdf), Section 4.1.6-4.1.7.)
>   * This is slightly more challenging with RSA but is still doable as [described here](https://crypto.stackexchange.com/a/26190) and [implemented here](https://gist.github.com/divergentdave/40ac9c7224b382166c905e76595bcf73).

Reply:
Ah sorry, I only looked at the diff, not the full file.